modules/configuration/pages/secrets.adoc

```diff
@@ -1,26 +1,68 @@
+== Store secrets in environment variables
 
-One of the most prolific approaches to providing secrets to a service is via environment variables. Redpanda Connect allows you to inject the values of environment variables into a configuration with the interpolation syntax `+${FOO}+`, within a configuration it looks like this:
+A common way to securely pass secrets to a service is to use environment variables. Redpanda Connect allows you to inject the values of environment variables into a configuration with the interpolation syntax `+${SECRET_ID}+`. For example:
 
 [source,yml]
 ----
 thing:
-  super_secret: "${SECRET}"
+  super_secret: "${SECRET_ID}"
 ----
 
 [NOTE]
 .Use quotes
 ====
-Note that it would be valid to have `+super_secret: ${SECRET}+` above (without the quotes), but if `SECRET` is unset then the configuration becomes structurally different. Therefore, it's always best to wrap environment variable interpolations with quotes so that when the variable is unset you still have a valid configuration (with an empty string).
+It is valid to have `+super_secret: ${SECRET}+` above (without the quotes), but if `SECRET` is unset then the configuration becomes structurally different. Therefore, it's always best to wrap environment variable interpolations with quotes so that when the variable is unset you still have a valid configuration (with an empty string).
 ====
 
-More information about this syntax can be found on the xref:configuration:interpolation.adoc[interpolation field page].
+For more information about this syntax, see the xref:configuration:interpolation.adoc[interpolation field page].
 
-== Using CLI flags
+== Look up secrets on a remote system at runtime
+
+From version 4.39.0, you can use the `rpk connect` CLI flag `--secrets` to:
+
+- Look up secrets values on a remote system at runtime
+- Set an environment variable value in your configuration
+
+For example, you could run the following command to retrieve the value for `"$\{SECRET}"`, when the secret is stored on a Redis server.
+
+```bash
+
+rpk connect run ./config.yaml --secrets env:redis://secret:redis@example.com
+
+```
+The `env:` argument tries to load the secret value from the specified URN using the format `scheme://prefix`. The prefix includes the environment variable name, and secret location, which in this case is the Redis server address.
+
+You can specify multiple URNs separated by commas, which are tried in turn until a secrets value is successfully returned.
+
+NOTE: If you use the argument `none:` with the `--secrets` flag, the secrets value is still returned, but Redpanda Connect does not set an environment variable.
+
+=== Supported remote systems
+
+You can retrieve secrets from all of the following remote systems.
+
+|===
+| Remote system | URN format
+
+| AWS
+| `aws://region/prefix`
+
+| AZURE
+| `az://vault-uri/prefix`. The `vault-uri` value should not contain a `https://` prefix.
+
+| GCP
+| `gcp://projectID/prefix`
+
+| Redis
+| `redis://prefix`
```

Comment:
See above. We do not support prefix for Redis. The URN should be of the form: `redis://secret:redis@example.com`

Reply:
Updated.